app/frontend/src/features/search/FilterDialog.tsx

```diff
@@ -76,6 +77,34 @@ export default function FilterDialog({
     onClose();
   });
 
+  //because we are abusing URLSearchParams as a kind of global mutable state, we need
+  //to calculate the default values and put them in a ref so they don't change
+  //otherwise we get console warnings about defaultValue changing
+  const defaultValues = useRef({
```

Comment:
This `URLSearchParams` thing feels "wrong", but it's pretty neat at the same time since the UI is still updating without any React state updates by embracing uncontrolled elements and using the URL as the eventual single source of state. I feel like I'm still wrapping my head around this 😄

Reply:
Yeah I completely agree.

I could change it to use a provider in `SearchPage`. It would cause a big re-render but I could keep the "pending" filters in a ref and "active" filters in state so it only re-renders when the dialog closes...

I think I will do that - it will be a lot more maintainable actually because the current way isn't very type-safe

Reply:
Unfortunately I still needed this bit to prevent the warning, but I think the refactor was worth it anyway :P